src/core/toga/sources/dicts.py

```diff
@@ -0,0 +1,73 @@
+from .base import Source, Node, Datum, to_accessor
+
+
+class BaseDictSource(Source):
+    def __init__(self, accessors):
+        super().__init__()
+        self._accessors = accessors
+
+
+class DictSource(BaseDictSource):
+    def __init__(self, accessors, data):
+        super().__init__(accessors)
+        self._roots = self.create_nodes(data)
+
+    def create_node(self, datum, children=None):
```

Comment:
Because we're writing our own tree instantiation class, I would _really_ love to see some tests for these methods when we get to writing tests.

Reply:
Naturally. This was more of a progress drop, to see if there was any reaction to the API changes I've proposed.